packages/mds-metrics/types.ts

```diff
@@ -19,30 +18,70 @@ export type EventSnapshotResponse = {
   [T in VEHICLE_TYPE]: { [S in VEHICLE_EVENT]: number }
 }
 
-export type CountsResponse = {
-  [T in VEHICLE_TYPE]: number
+export type TelemetryCountsResponse = {
+  telemetryCount: {
+    count: number
+    slacount: number
+    provider_id: UUID
+  }[]
+  slice: {
+    start: number
+    end: number
+  }
+}
+
+export type EventCountsResponse = {
+  eventCount: {
+    count: number
+    slacount: number
+    event_type: string
+    provider_id: UUID
+  }[]
+  slice: {
+    start: number
+    end: number
+  }
+}
+
+export type GetTimeBinsParams = {
+  start_time: Timestamp
+  end_time: Timestamp
+  bin_size: Timestamp
+}
+export interface MetricsApiRequest extends ApiRequest {
+  query: Partial<
+    {
+      [P in 'start_time' | 'end_time' | 'bin']: string
+    }
+  >
```

Comment:
It looks like you are getting the data from the json request body and not the query string.
So, this can be:
```
export interface MetricsApiRequest extends ApiRequest {
  body: Partial<GetTimeBinsParams>
}